File: Julia/Root Estimation/newtonraphson.jl
function newtonRaphson(f::Function, fprime::Function, xi::Number, ea::Number, maxIter::Int64)
  # Estimate roots of functions using the Newton-Raphson method
  for i = 1:maxIter
    xiprev  = xi;
    xi = xi - (f(xi) / fprime(xi));

    if abs(xi - xiprev) / xi < ea
      break;
    end;
  end
  return xi;
end

File: MATLAB/Root Estimation/newtonraphson.m
function [xr] = newtonraphson(f, fprime, xi, ea, maxIter)
%newtonraphson Estimate roots using the Newton-Raphson method
%   xr = newtonraphson(f,fprime,xi,ea,maxIter) estimates the root xr of the
%   anonymous function f using its derivative fprime and an initial
%   estimate of xi. It iterates on the estimate until the relative error
%   falls below ea or maxIter iterations occur.
  xr = xi;
  for ii = 1:maxIter
    xrprev = xr;
    xr = xr - (f(xr)/fprime(xr));
    
    if abs(xr - xrprev)/xr < ea
      break;
    end
  end
end
